java.lang.ClassCastException on server start

**Describe the bug**  
`./server run badapp`
Server did not start and a ClassCastException was thrown

```
Launching badapp (Open Liberty 25.0.0.6/wlp-1.0.102.cl250620250602-1102) on Eclipse OpenJ9 VM, version 17.0.11+9 (en_CA)
CWWKE0005E: The runtime environment could not be launched.
CWWKE0018E: An exception occurred while launching the runtime environment: java.lang.ClassCastException: com.ibm.ws.logging.internal.impl.BaseTraceService$SystemLogHolder incompatible with com.ibm.ws.logging.utils.FileLogHolder
com.ibm.ws.kernel.boot.LaunchException: Caught unexpected exception java.lang.ClassCastException: com.ibm.ws.logging.internal.impl.BaseTraceService$SystemLogHolder incompatible with com.ibm.ws.logging.utils.FileLogHolder
	at com.ibm.ws.kernel.boot.internal.KernelBootstrap.rethrowException(KernelBootstrap.java:714)
	at com.ibm.ws.kernel.boot.internal.KernelBootstrap.go(KernelBootstrap.java:221)
	at com.ibm.ws.kernel.boot.Launcher.handleActions(Launcher.java:243)
	at com.ibm.ws.kernel.boot.Launcher.createPlatform(Launcher.java:119)
	at com.ibm.ws.kernel.boot.cmdline.EnvCheck.main(EnvCheck.java:61)
	at com.ibm.ws.kernel.boot.cmdline.EnvCheck.main(EnvCheck.java:37)
Caused by: java.lang.ClassCastException: com.ibm.ws.logging.internal.impl.BaseTraceService$SystemLogHolder incompatible with com.ibm.ws.logging.utils.FileLogHolder
	at com.ibm.ws.logging.internal.impl.BaseTraceService$TimedLogRoller.<init>(BaseTraceService.java:2252)
	at com.ibm.ws.logging.internal.impl.BaseTraceService.scheduleTimeBasedLogRollover(BaseTraceService.java:1520)
	at com.ibm.ws.logging.internal.impl.BaseTraceService.update(BaseTraceService.java:436)
	at com.ibm.ws.logging.internal.impl.BaseTraceService.init(BaseTraceService.java:356)
	at com.ibm.websphere.ras.TrConfigurator.init(TrConfigurator.java:90)
	at com.ibm.ws.logging.internal.impl.LogProviderImpl.configure(LogProviderImpl.java:35)
	at com.ibm.ws.kernel.launch.internal.LauncherDelegateImpl.getLogProviderImpl(LauncherDelegateImpl.java:197)
	at com.ibm.ws.kernel.launch.internal.LauncherDelegateImpl.launchFramework(LauncherDelegateImpl.java:93)
	at com.ibm.ws.kernel.boot.internal.KernelBootstrap.go(KernelBootstrap.java:215)
	... 4 more

```


**Steps to Reproduce**  

`bootstrap.properties`
```
com.ibm.ws.logging.trace.file.name=stdout
```

`server.env`
```
WLP_LOGGING_ROLLOVER_START_TIME=00:00
WLP_LOGGING_ROLLOVER_INTERVAL=1m
```

`server.xml`
```
<?xml version="1.0" encoding="UTF-8"?>
<server description="new server">

    <!-- Enable features -->
    <featureManager>
        <feature>servlet-6.0</feature>
        <feature>restfulWS-3.1</feature>
        <feature>monitor-1.0</feature>
    </featureManager>

    <!-- To access this server from a remote client add a host attribute to the following element, e.g. host="*" -->
    <httpEndpoint id="defaultHttpEndpoint"
                  httpPort="9080"
                  httpsPort="9443" />

    <!-- Automatically expand WAR files and EAR files -->
    <applicationManager autoExpand="true"/>

    <!-- Default SSL configuration enables trust for default certificates from the Java runtime --> 
    <ssl id="defaultSSLConfig" trustDefaultCerts="true" />

    <webApplication contextRoot="/badapp" location="badapp.war"/>

    <logging traceSpecification="com.ibm.ws.lumberjack.badness.*=all:com.ibm.ws.webcontainer.*=all" />
</server>
```



**Expected behavior**  
A clear and concise description of what you expected to happen.

**Diagnostic information:**  
 - OpenLiberty Version: 25.0.0.6
 - Affected feature(s): 
 - Java Version: IBM Semeru Runtime Open Edition (17.0.11+9)
 - server.xml configuration (WITHOUT sensitive information like passwords)
 - If it would be useful, upload the messages.log file found in `$WLP_OUTPUT_DIR/messages.log`

**Additional context**  
Add any other context about the problem here.

--- a/dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java
+++ b/dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java
@@ -242,6 +242,9 @@ public class BaseTraceService implements TrService {
     /** The maximum FFDC file age before deletion. */
     private volatile long maxFfdcAge = -1;
 
+    /** The trace file name */
+    private volatile String traceFileName = "trace.log";
+
     /** Early msgs issued before MessageRouter is started. */
     protected volatile Queue<RoutedMessage> earlierMessages = new SimpleRotatingSoftQueue<RoutedMessage>(new RoutedMessage[100]);
     protected volatile Queue<RoutedMessage> earlierTraces = new SimpleRotatingSoftQueue<RoutedMessage>(new RoutedMessage[200]);
@@ -1433,9 +1436,9 @@ public class BaseTraceService implements TrService {
             //null and empty rolloverStartTime are the same
             if (rolloverStartTime == null)
                 rolloverStartTime = "";
-            // If neither of the rollover attributes change, return without rescheduling
-            // Also, if the traceFileName is set to "stdout", then we must reschedule the rollover by omitting the trace.log file.
-            if (this.rolloverStartTime.equals(rolloverStartTime) && this.rolloverInterval == rolloverInterval && !traceFileName.equals("stdout")) {
+            // If neither of the rollover attributes change, return without rescheduling.
+            // Also, if the traceFileName attribute value did NOT change, return without rescheduling
+            if (this.rolloverStartTime.equals(rolloverStartTime) && this.rolloverInterval == rolloverInterval && this.traceFileName.equals(traceFileName)) {
                 return;
             } else {
                 timedLogRollover_Timer.cancel();
@@ -1480,6 +1483,7 @@ public class BaseTraceService implements TrService {
 
         this.rolloverStartTime = rolloverStartTime;
         this.rolloverInterval = rolloverInterval;
+        this.traceFileName = traceFileName; // Preserve the previous traceFileName, in order to check if the attribute changed.
 
         //parse startTimeField
         String[] hourMinPair = rolloverStartTime.split(":");

--- a/dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java
+++ b/dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java
@@ -1425,16 +1425,17 @@ public class BaseTraceService implements TrService {
     private void scheduleTimeBasedLogRollover(LogProviderConfigImpl config) {
         String rolloverStartTime = config.getRolloverStartTime();
         long rolloverInterval = config.getRolloverInterval();
-        String fileName = config.getTraceFileName();
+        String traceFileName = config.getTraceFileName();
 
-        //if the rollover has already been scheduled, cancel it
-        //this is either a reschedule, or a unschedule
+        // If the rollover has already been scheduled, cancel it
+        // This is either a reschedule, or a unschedule
         if (this.isLogRolloverScheduled) {
             //null and empty rolloverStartTime are the same
             if (rolloverStartTime == null)
                 rolloverStartTime = "";
-            //if neither of the rollover attributes change, return without rescheduling
-            if (this.rolloverStartTime.equals(rolloverStartTime) && this.rolloverInterval == rolloverInterval) {
+            // If neither of the rollover attributes change, return without rescheduling
+            // Also, if the traceFileName is set to "stdout", then we must reschedule the rollover by omitting the trace.log file.
+            if (this.rolloverStartTime.equals(rolloverStartTime) && this.rolloverInterval == rolloverInterval && !traceFileName.equals("stdout")) {
                 return;
             } else {
                 timedLogRollover_Timer.cancel();
@@ -1519,7 +1520,7 @@ public class BaseTraceService implements TrService {
         //schedule rollover
         timedLogRollover_Timer = new Timer(true);
         TimedLogRoller tlr;
-        if (fileName.equals("stdout")) {
+        if (traceFileName.equals("stdout")) {
             // If traceFileName is configured to stdout, that means there will be no trace.log to rollover,
             // omit the trace.log, when scheduling the rollover.
             tlr = new TimedLogRoller(messagesLog);

--- a/dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java
+++ b/dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java
@@ -1425,6 +1425,7 @@ public class BaseTraceService implements TrService {
     private void scheduleTimeBasedLogRollover(LogProviderConfigImpl config) {
         String rolloverStartTime = config.getRolloverStartTime();
         long rolloverInterval = config.getRolloverInterval();
+        String fileName = config.getTraceFileName();
 
         //if the rollover has already been scheduled, cancel it
         //this is either a reschedule, or a unschedule
@@ -1517,7 +1518,14 @@ public class BaseTraceService implements TrService {
         }
         //schedule rollover
         timedLogRollover_Timer = new Timer(true);
-        TimedLogRoller tlr = new TimedLogRoller(messagesLog, traceLog);
+        TimedLogRoller tlr;
+        if (fileName.equals("stdout")) {
+            // If traceFileName is configured to stdout, that means there will be no trace.log to rollover,
+            // omit the trace.log, when scheduling the rollover.
+            tlr = new TimedLogRoller(messagesLog);
+        } else {
+            tlr = new TimedLogRoller(messagesLog, traceLog);
+        }
         timedLogRollover_Timer.scheduleAtFixedRate(tlr, firstRollover, rolloverInterval * 60000);
         this.isLogRolloverScheduled = true;
     }
@@ -2252,10 +2260,16 @@ public class BaseTraceService implements TrService {
             flhTrace = (FileLogHolder) trace;
         }
 
+        TimedLogRoller(TraceWriter messages) {
+            flhMessages = (FileLogHolder) messages;
+            // This means when traceFileName="stdout", there will be no trace.log file to rollover.
+            flhTrace = null;
+        }
+
         @Override
         public void run() {
             flhMessages.createStream(true);
-            if (TraceComponent.isAnyTracingEnabled()) {
+            if (flhTrace != null && TraceComponent.isAnyTracingEnabled()) {
                 flhTrace.createStream(true);
             }
         }

